### PR TITLE
Add new option --nobloom.

### PR DESCRIPTION
**Pull request by dajohi**
Add a new configuration option --nobloom that disables bloom
filtering (BIP0037) support.  Bloom filtering is very CPU
intensive, so a user might wish to specify this option to lower
CPU utilization at the cost of increased bandwidth usage.  However,
this option also increases both CPU utilization and bandwidth usage
for SPV clients.

refs #172

### CODE REVIEW

File: peer.go [hunk 1]
```diff
@@ -554,10 +554,29 @@ func (p *peer) pushMerkleBlockMsg(sha *wire.ShaHash, doneChan, waitChan chan str
...
+		// original block.
+		header := blk.MsgBlock().Header
+		numTxs := len(blk.Transactions())
+
+		merkle = wire.NewMsgMerkleBlock(&header)
+		merkle.Transactions = uint32(numTxs)
+		for _, tx := range blk.Transactions() {
+			merkle.Hashes = append(merkle.Hashes, tx.Sha())
+		}
+		merkle.Flags = bytes.Repeat([]byte{0x01}, numTxs)
```
Comment (davecgh) [inline peer.go]: This is not correct.  That Flags parameter of a merkle block is a bitset.  This should therefore choosing the correct number of bytes based on the number of bits needed to represent each transaction and setting trailing zeros in the final byte for transactions which are not present.

**Comment by davecgh** [comment]
This isn't ready yet since the merkle block being served currently has invalid flags.

**Comment by davecgh** [comment]
I'd rather wait until `SFNodeBloom` service bit is in place along with the protocol version bump and morph this pull request into working properly with that.

**Comment by dajohi** [comment]
Close in favor of #499.

**Pull request closed without merging**